integrations/tensorflow/test/requirements.txt

```diff
@@ -1,15 +1,4 @@
 # Requirements for running TF tests
-tf-nightly==2.17.0.dev20240507
+tensorflow>=2.16.1
```

Comment:
> Separately we can update documentation (https://iree.dev/guides/ml-frameworks/tensorflow/#prerequisites) and samples (Colab notebooks) to use stable instead of nightly.

https://colab.research.google.com/github/iree-org/iree/blob/main/samples/colab/tensorflow_edge_detection.ipynb at least still needs a newer TF. Colab is currently on 2.15.0, which fails with
```
CompilerToolError: Error invoking IREE compiler tool iree-compile
Error code: 1
Diagnostics:
-:5:10: error: 'stablehlo.convolution' op attribute 'window_strides' failed to satisfy constraint: DenseI64ArrayAttr with generic Attribute storage
-:5:10: note: in bytecode version 1 produced by: MLIR18.0.0git
```

2.16.1 (latest, installed here) works

(should get the tools using VHLO instead of StableHLO...)